test/ColorLineTracingTest.cpp

```diff
@@ -0,0 +1,284 @@
+/**
+ * @file   ColorLineTracingTest.cpp
+ * @brief  ColorLineTracingクラスのテスト
+ * @author CHIHAYATAKU
+ */
+
+#include "ColorLineTracing.h"
+#include <gtest/gtest.h>
+#include <gtest/internal/gtest-port.h>
+
+using namespace std;
+
+namespace etrobocon2024_test {
+  // 最初3回の色取得で連続して指定色を取得するテストケース
+  TEST(ColorLineTracingTest, runToGetFirst)
+  {
+    Controller controller;
+    Measurer measurer;
+    // PWMの初期化
+    controller.resetWheelsMotorPwm();
+    COLOR targetColor = COLOR::GREEN;
+    double targetSpeed = 50000.0;
+    double targetBrightness = 50.0;
+    PidGain gain = { 0.1, 0.05, 0.05 };
+    bool isLeftEdge = true;
+    ColorLineTracing cl(targetColor, targetSpeed, targetBrightness, gain, isLeftEdge);
+
+    // 初期値から期待する走行距離を求める
+    // int initialRightCount = measurer.getRightCount();
+    // int initialLeftCount = measurer.getLeftCount();
+    // double expected = Mileage::calculateMileage(initialRightCount, initialLeftCount);
+    double expected = 0.0;
+    srand(9037);  // 3回連続して緑を取得する乱数シード
+    cl.run();     // 緑までライントレースを実行
+
+    // ライントレース後の走行距離
+    int rightCount = measurer.getRightCount();
+    int leftCount = measurer.getLeftCount();
+    double actual = Mileage::calculateMileage(rightCount, leftCount);
+
+    EXPECT_LT(expected, actual);  // 初期値より少しでも進んでいる
+  }
+
+  // 少し走ってから指定色を取得するテストケース
```

Comment:
```suggestion
  // 少し走ってから指定色を取得するテストケース (左エッジ走行)
```
細かいしコメントのところだけど、下と同じテストケース名になってる。
runLeftEdge, runRightEdge の意味を汲み取れるように変更お願いします🙏